### ISSUE TITLE
x/build: "unexpected stale targets" on darwin builders

### ISSUE THREAD

**Issue by bcmills** [golang/go#33598]
Two flakes in the `darwin-amd64-10_11` builders over the past week or so, both with the same symptom: `unexpected stale targets` failures for `internal/cpu`, and a `build ID mismatch` for `runtime/internal/sys`.

* https://build.golang.org/log/e83a781b6c57b39a62a8802c6e935e2580d7d6d5
* https://build.golang.org/log/3edfb9dcd284c38a9ac3c7c4d678feca4a3042eb

CC @jayconrod @ianlancetaylor @aclements 

<details>

```
darwin-amd64-10_11 at 0ca4f6be35d2abf54128b57c3aa1a1fcbd5adfe3

:: Running /var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/make.bash with args ["/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/make.bash"] and env ["TERM_PROGRAM=Apple_Terminal" "TERM=xterm-256color" "SHELL=/bin/bash" "TMPDIR=/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/" "Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.55fT0Y9EVR/Render" "TERM_PROGRAM_VERSION=361.1" "TERM_SESSION_ID=D33F650A-CE77-4BF0-B95E-A37406DCC122" "USER=gopher" "SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.43nQd98RPC/Listeners" "__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0" "PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin" "PWD=/Users/gopher" "LANG=en_US.UTF-8" "XPC_FLAGS=0x0" "XPC_SERVICE_NAME=0" "SHLVL=2" "HOME=/Users/gopher" "LOGNAME=gopher" "SECURITYSESSIONID=186a8" "GO_BUILDER_ENV=macstadium_vm" "GO_STAGE0_NET_DELAY=300ms" "GO_STAGE0_DL_DELAY=0s" "WORKDIR=/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir" "GOROOT_BOOTSTRAP=/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go1.4" "GO_BUILDER_NAME=darwin-amd64-10_11" "GOROOT_BOOTSTRAP=/Users/gopher/go1.4" "GOBIN=" "TMPDIR=/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/tmp" "GOCACHE=/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/gocache"] in dir /var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src

Building Go cmd/dist using /Users/gopher/go1.4.
Building Go toolchain1 using /Users/gopher/go1.4.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
HASH[build runtime/internal/sys]
HASH[build runtime/internal/sys]: "devel 0ca4f6be35d2abf54128b57c3aa1a1fcbd5adfe3"
HASH[build runtime/internal/sys]: "compile\n"
HASH[build runtime/internal/sys]: "goos darwin goarch amd64\n"
HASH[build runtime/internal/sys]: "import \"runtime/internal/sys\"\n"
HASH[build runtime/internal/sys]: "omitdebug false standard true local false prefix \"\"\n"
HASH[build runtime/internal/sys]: "modinfo \"\"\n"
HASH[build runtime/internal/sys]: "compile PRWJHAzAczrrtWSoyeY4 [] []\n"
HASH[build runtime/internal/sys]: "=\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/arch.go: d9b0b7e72538d421b2607acaba60ca49f20ef584b3d1d191c6729e35fbb8101d
HASH[build runtime/internal/sys]: "file arch.go 2bC35yU41CGyYHrKumDK\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/arch_amd64.go: d301f8642ec7fcc35d0d39a4b07e0187971afea7c6d0dfd5ddcef1d031670160
HASH[build runtime/internal/sys]: "file arch_amd64.go 0wH4ZC7H_MNdDTmksH4B\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/intrinsics.go: 8b469a461e1d983706e0b3635715ce70691adc5db7c4e067b88cc59f40cd66f4
HASH[build runtime/internal/sys]: "file intrinsics.go i0aaRh4dmDcG4LNjVxXO\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/stubs.go: 23b3e5c631b086fe7a2dec4bf044600e034bf6a8eeb25e0a19efc4ce6311423d
HASH[build runtime/internal/sys]: "file stubs.go I7PlxjGwhv56LexL8ERg\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/sys.go: 55e021891200a7e6a5c371c8a1ab71b6c15aeb16ea6c1b192185d17df8c8b18f
HASH[build runtime/internal/sys]: "file sys.go VeAhiRIAp-alw3HIoatx\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/zgoarch_amd64.go: 6f2fa4ebe8999a3b7cfe3ead6e24b8356ed842292f23cb1b4f995c0b5b45126b
HASH[build runtime/internal/sys]: "file zgoarch_amd64.go by-k6-iZmjt8_j6tbiS4\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/zgoos_darwin.go: 464151a4ab63f8c7aab345f457789cb12c9b68051fc73b8a675717d7886214dc
HASH[build runtime/internal/sys]: "file zgoos_darwin.go RkFRpKtj-Meqs0X0V3ic\n"
HASH /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/src/runtime/internal/sys/zversion.go: 623491756e2adacebf4c53078883e4284349c6159d0404aba9975887d7660d06
HASH[build runtime/internal/sys]: "file zversion.go YjSRdW4q2s6_TFMHiIPk\n"
HASH[build runtime/internal/sys]: cd175367cc0e62c53ed8e20197f8e44b2270f61d3eb7b4f3c1991a5abb06ae16
runtime/internal/sys true
go tool dist: unexpected stale targets reported by /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir/go/pkg/tool/darwin_amd64/go_bootstrap list -gcflags="" -ldflags="" for [cmd/asm cmd/cgo cmd/compile cmd/link runtime/internal/sys]:
	STALE cmd/asm: stale dependency: internal/cpu
	STALE cmd/cgo: stale dependency: internal/cpu
	STALE cmd/compile: stale dependency: internal/cpu
	STALE cmd/link: stale dependency: internal/cpu
	STALE runtime/internal/sys: build ID mismatch
```

</details>

**Comment by bcmills** [comment]
`darwin-amd64-10_14`: https://build.golang.org/log/b56f0ee52abc686e36bc921be252232bbf7afbd9

**Comment by bcmills** [comment]
Not clear to me whether this is during toolchain3, or if the underlying cause is even related, but there is a similar symptom on the `linux-ppc64le-power9osu` builder in https://build.golang.org/log/57277f725843345df871d34cfabf1fda58934be0.

**Comment by laboger** [comment]
I see the same type of failure with the stale targets when I try to force -pie to be the default build mode for ppc64le (due to a request). It is noted in https://github.com/golang/go/issues/28531. Maybe something wrong with the way that internal/cpu is hashed so that it doesn't get consistent results?

**Comment by bcmills** [comment]
[2019-11-11T17:53:51-70be481/darwin-amd64-10_11](https://build.golang.org/log/3459f18c1f6fe237f0f30c07ab0e20023a8ee192)
[2019-11-07T18:39:03-05aa4a7/darwin-amd64-10_11](https://build.golang.org/log/0acb290141c3b03d1d0d6a570a91e8327c380819)
[2019-11-07T08:25:32-7a2baa9/darwin-amd64-10_11](https://build.golang.org/log/259722ea1171a52202216a1f272576c54f70195b)
[2019-10-31T01:09:14-851b1f4/illumos-amd64](https://build.golang.org/log/d6a510b7b2061693603e2fde70b087744a6792c9)
[2019-10-31T01:09:14-851b1f4/solaris-amd64-oraclerel](https://build.golang.org/log/0be2ab048cb1722f3e2c72cdec3bc593cf22bbf7)
[2019-10-31T01:04:59-e90e6e7/illumos-amd64](https://build.golang.org/log/3f885a454658db207471c2056d4fa06df05a319f)
[2019-10-31T01:04:59-e90e6e7/solaris-amd64-oraclerel](https://build.golang.org/log/13c507aeed0a718706edfb3c0472bd34a4ea8907)
[2019-10-25T17:41:36-d77b809/illumos-amd64](https://build.golang.org/log/c0e47740c703cb7c53b954619e5c2b28e8536df4)
[2019-10-25T17:41:36-d77b809/solaris-amd64-oraclerel](https://build.golang.org/log/3c104b2b198524421047a225180a43b50503941d)
[2019-10-25T17:15:50-2e2ef66/illumos-amd64](https://build.golang.org/log/8bf3587c4821937811c7d276567c1e3c241a9510)
[2019-10-25T17:15:50-2e2ef66/solaris-amd64-oraclerel](https://build.golang.org/log/6f48ed40653b9e5369281ec47dac90e2a6edbd6d)
[2019-10-25T16:51:18-376ef73/illumos-amd64](https://build.golang.org/log/f971152c3c265019903a76c42ee8d91bee0d2f12)
[2019-10-25T16:51:18-376ef73/solaris-amd64-oraclerel](https://build.golang.org/log/cd27b05c26c1b9d996d7aab75a6bb3ce8e45581b)
[2019-10-24T15:24:20-fb06609/illumos-amd64](https://build.golang.org/log/d6ea4f17ab366ceaac68b327f1b2a0edf0795091)
[2019-10-24T15:24:20-fb06609/solaris-amd64-oraclerel](https://build.golang.org/log/645677b4d0d4c9da93ae521badbd6a2e6b47fb74)
[2019-10-24T15:22:31-df01b79/illumos-amd64](https://build.golang.org/log/e8928b2b3021bada0493acae41b4047976acb32d)
[2019-10-24T15:22:31-df01b79/solaris-amd64-oraclerel](https://build.golang.org/log/977c1bbba7f547d33e59d0f1b5e30993a2455681)
[2019-10-24T00:53:39-4adf822/illumos-amd64](https://build.golang.org/log/6512a500e9a50c30243a02716da3a02bec480296)
[2019-10-24T00:53:39-4adf822/solaris-amd64-oraclerel](https://build.golang.org/log/7b013e361002e606bab3b06abbe2dc84b99fb6bd)
[2019-10-21T21:58:03-c480d32/illumos-amd64](https://build.golang.org/log/bf4b52ded549a9c78c394600873e4a817c7c3119)
[2019-10-21T21:58:03-c480d32/solaris-amd64-oraclerel](https://build.golang.org/log/81c7ae807576f53ac5977415a4249961ee935832)
[2019-10-21T21:57:32-d56c814/illumos-amd64](https://build.golang.org/log/f78adebe9420eb3f4494789d64dbd5dc1d839641)
[2019-10-21T21:57:32-d56c814/solaris-amd64-oraclerel](https://build.golang.org/log/4cd7648621b961b768e144592821ccc08cb5f51e)
[2019-10-21T21:57:10-4fa524b/illumos-amd64](https://build.golang.org/log/c5e79d162a0c4cf74ba884f8021afb3564344210)
[2019-10-21T21:57:10-4fa524b/solaris-amd64-oraclerel](https://build.golang.org/log/b2802cae956522a60ddcae2e86d764db34f14eab)
[2019-10-21T21:57:01-97e497b/illumos-amd64](https://build.golang.org/log/3f848830cae581bf69d87aabb3372822b004b765)
[2019-10-19T15:18:32-5777ffd/solaris-amd64-oraclerel](https://build.golang.org/log/c48fea9699f99acc0832aacb50247fc5b100a06d)
[2019-10-18T18:44:05-c3459ea/solaris-amd64-oraclerel](https://build.golang.org/log/e5dc390097dda0c4aef0998488892c493b391438)
[2019-10-17T20:28:28-bd22993/solaris-amd64-oraclerel](https://build.golang.org/log/8713cbbe26ac970100347219ca66b16de1ee2e11)
[2019-10-17T20:28:15-286d744/illumos-amd64](https://build.golang.org/log/7f6e13d3cb5b9a6d266eabda5c15fe623daa925a)
[2019-10-17T20:28:15-286d744/solaris-amd64-oraclerel](https://build.golang.org/log/d65e1260d8fef574d384ce04e318a28bec4dfe15)
[2019-10-17T20:08:33-b661547/illumos-amd64](https://build.golang.org/log/529fc7a064faf80064bf4770c8cf925d4ce72dd5)
[2019-10-17T20:08:33-b661547/solaris-amd64-oraclerel](https://build.golang.org/log/4898c2edff75b507c489e24f6018b734866cf6c3)
[2019-10-17T14:20:08-e5acb58/solaris-amd64-oraclerel](https://build.golang.org/log/3dd1e2cbb08b4669eb79075c99445a2c9a2b56ee)
[2019-10-17T14:20:00-6cbf37b/solaris-amd64-oraclerel](https://build.golang.org/log/eb5423a49d712399c26d72f78e9b2b9157ccd9b9)
[2019-10-17T14:18:37-8011051/solaris-amd64-oraclerel](https://build.golang.org/log/3dec37e9d60df97d82e3c0640aedd0c7649533f1)
[2019-10-17T14:18:28-15634a0/solaris-amd64-oraclerel](https://build.golang.org/log/01123177671b79bb9178ee8bb0b547ceb1ddf5b9)
[2019-10-16T22:28:18-2bbf2e0/solaris-amd64-oraclerel](https://build.golang.org/log/bc7ab74756fa34bb6e1e826d9254c6d7329b0517)
[2019-10-16T15:57:07-5caac2f/solaris-amd64-oraclerel](https://build.golang.org/log/d5d3df5abdc7655c7f4739ac061ec7b8e12086cd)
[2019-10-16T13:24:55-c9470b0/solaris-amd64-oraclerel](https://build.golang.org/log/168e08dc70fa9703efe13afaacca4219960b9dea)
[2019-10-15T19:00:07-4a9b30c/solaris-amd64-oraclerel](https://build.golang.org/log/1c659b06f179f80ac35b35040ea117f02fccd61e)
[2019-10-15T19:00:00-7256f50/solaris-amd64-oraclerel](https://build.golang.org/log/6a3b69407fe727d3525be8d06f10711d02a32683)
[2019-10-15T18:59:52-5fec788/solaris-amd64-oraclerel](https://build.golang.org/log/7706ff33d835d4384a965ef9a3b1e92fdee33eaa)
[2019-10-15T18:56:08-6ba3ae9/solaris-amd64-oraclerel](https://build.golang.org/log/7baf731ffc52b07e419cb7d6ee65a648a87eddfc)
[2019-10-15T18:55:59-dab05a0/solaris-amd64-oraclerel](https://build.golang.org/log/e665f3e565d8e769e01577c2f600501495d934f8)
[2019-10-15T18:55:48-27111e5/solaris-amd64-oraclerel](https://build.golang.org/log/0bee3e370adecabbb4793f481d73ad9f46415e3b)
[2019-10-15T18:55:32-cab29eb/solaris-amd64-oraclerel](https://build.golang.org/log/a988479bcbbb3f50341aeb2c48faa46e75e19648)
[2019-09-11T12:19:26-11c2411/darwin-amd64-10_14](https://build.golang.org/log/b56f0ee52abc686e36bc921be252232bbf7afbd9)
[2019-08-31T00:47:27-e5e5fb0/darwin-amd64-10_11](https://build.golang.org/log/a95eb2d66fee5aa876d81a17ddb1e861704c71b8)
[2019-08-26T19:39:05-97edf77/darwin-amd64-nocgo](https://build.golang.org/log/51522ce78248f3033ac4106c9c8aaed15b169b38)
[2019-08-08T16:16:15-f4be93a/darwin-amd64-10_11](https://build.golang.org/log/e83a781b6c57b39a62a8802c6e935e2580d7d6d5)
[2019-08-07T14:40:17-0ca4f6b/darwin-amd64-10_11](https://build.golang.org/log/3edfb9dcd284c38a9ac3c7c4d678feca4a3042eb)
[2019-07-01T21:30:23-d410642/darwin-amd64-10_12](https://build.golang.org/log/ec4b0bab6d0036490401b4c76c1485e445d7558a)

**Comment by bcmills** [comment]
@laboger, the `ppc64le` failures seem to occur at a different point in the build process. I'll file a separate issue for that.

**Comment by bcmills** [comment]
[2020-02-19T21:26:10-7299308/darwin-amd64-10_12](https://build.golang.org/log/9af138c862369538f454fb5b8e239444bf9febe2)

**Comment by bcmills** [comment]
Curiously, the only builder that has failed in this way in the last month or so is the `plan9-arm` builder.

(There hasn't been another `darwin-amd64` failure since February, and we have a bit of circumstantial evidence from linker crashes that that rate of failure may be due to platform filesystem instability.)

[2020-03-02T17:40:11-a2bff7c/plan9-arm](https://build.golang.org/log/e48c917bfdbe4f9d3f55a7a3fbdacb96906e489e)
[2020-02-19T21:26:10-7299308/darwin-amd64-10_12](https://build.golang.org/log/9af138c862369538f454fb5b8e239444bf9febe2)
[2020-02-18T19:44:25-a0cf2c8/plan9-arm](https://build.golang.org/log/fe4b647df24cd4e4193ebcd8708459ea6ed25071)

**Comment by bcmills** [comment]
Curiously, [CL 223755](https://golang.org/cl/223755) seems to demonstrate that the `darwin-amd64-10_12` builder is receiving a stale snapshot (or not using snapshots and not rebuilding appropriately).

```
Building Go cmd/dist using /Users/gopher/go1.4. (go1.4-bootstrap-20161024 darwin/amd64)
Building Go toolchain1 using /Users/gopher/go1.4.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for darwin/amd64.
---
Installed Go for darwin/amd64 in /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir-host-darwin-10_12/go
Installed commands in /private/var/folders/dx/k53rs1s93538b4x20g46cj_w0000gn/T/workdir-host-darwin-10_12/go/bin
…
go tool dist: unexpected stale targets reported by go list -gcflags="" -ldflags="" for [std cmd]:
	STALE cmd/addr2line: build ID mismatch
	STALE cmd/api: build ID mismatch
	STALE cmd/asm: build ID mismatch
	STALE cmd/buildid: build ID mismatch
	STALE cmd/cgo: build ID mismatch
	STALE cmd/compile: build ID mismatch
	STALE cmd/cover: build ID mismatch
	STALE cmd/dist: build ID mismatch
	STALE cmd/doc: build ID mismatch
	STALE cmd/fix: build ID mismatch
	STALE cmd/go: build ID mismatch
	STALE cmd/gofmt: build ID mismatch
	STALE cmd/link: build ID mismatch
	STALE cmd/nm: build ID mismatch
	STALE cmd/objdump: build ID mismatch
	STALE cmd/pack: build ID mismatch
	STALE cmd/pprof: build ID mismatch
	STALE cmd/test2json: build ID mismatch
	STALE cmd/trace: build ID mismatch
	STALE cmd/vet: build ID mismatch
```

CC @cagedmantis @dmitshur @toothrot

**Comment by bcmills** [comment]
I haven't seen the `solaris` failure mode in months, so retitling to focus on the remaining `darwin` failures.

**Comment by bcmills** [comment]
The affected builders seem to be:
* `darwin-amd64-10_12`
* `darwin-amd64-10_14`
* `darwin-amd64-10_15`

Curiously, `darwin-386-10_14` and `darwin-amd64-nocgo` appear to be unaffected. To me, that suggests a problem with the builder configuration rather than a bug in `cmd/dist`.

**Comment by bcmills** [comment]
(Those builders broke due to unexpected staleness in [CL 223746](https://golang.org/cl/223746).)

**Comment by toothrot** [comment]
This could be related to #35459 /cc @cagedmantis

**Comment by bcmills** [comment]
`darwin-386-10_14` and `darwin-amd64-nocgo` _seemed_ unaffected only because they skipped the affected test for some reason.

`darwin-amd64-race` does not detect `cmd` staleness because for some reason the `shouldTestCmd` function in `cmd/dist` exempts the `cmd` binaries from race-mode checking. That seems unwise, but it's not the topic of this issue.

**Comment by bcmills** [comment]
(Filed the missing `race` coverage as #37940.)

**Comment by bcmills** [comment]
Another clue, from [CL 236819](https://golang.org/cl/236819): if we do not set `GOROOT_FINAL_OLD` in `cmd/dist`, then `cmd/link/dwarf_test.go` also fails due to `cmd/link` being stale.

https://storage.googleapis.com/go-build-log/b83e1d59/darwin-amd64-10_14_cf83aeff.log

**Comment by randall77** [comment]
Just happened on a trybot build, CL 238077, trybot run on patchset 6. linux/arm.

**Comment by bcmills** [comment]
[2021-01-11T19:45:02-7593090/darwin-arm64-11_0-toothrot](https://build.golang.org/log/44e8515172658a8cee5526238d2e3860cfb2b14b)
[2021-01-01T20:05:20-3dd5867/darwin-arm64-11_0-toothrot](https://build.golang.org/log/f7e1d92bbe7892e6dd6bf6c574bfcc38a97ddd8d)
[2020-10-16T17:31:17-83317d9/darwin-amd64-10_14](https://build.golang.org/log/d3fa33dc3957ea7504e3551514049bbac0c967e3)
[2020-10-01T17:48:42-8c84dcf/darwin-amd64-race](https://build.golang.org/log/5d2994f98a6189516c1ed6d7af89fc3b83fefda5)
[2020-09-01T19:10:59-829ca10/darwin-amd64-10_12](https://build.golang.org/log/1c543eb516ad8a36f8cddb04eed61ab8f0a689a9)

**Comment by bcmills** [comment]
Looking at the latest test failure from [CL 223755](https://golang.org/cl/223755), I see:
```
Installed Go for darwin/amd64 in /private/var/folders/9w/4l2_g3kx01x199n37fbmv3s80000gn/T/workdir-host-darwin-10_14/go
```

That message comes from here, and implies that `GOROOT` is `/private/…`:
https://github.com/golang/go/blob/b1bedc0774d8a3a7ff8778e933ee92e8638e9493/src/cmd/dist/build.go#L1707-L1708

But when I `gomote ssh` into an instance, I see:
```
# `gomote push` and the builders use:
# - workdir: /var/folders/9w/4l2_g3kx01x199n37fbmv3s80000gn/T/workdir-host-darwin-10_14
# - GOROOT: /var/folders/9w/4l2_g3kx01x199n37fbmv3s80000gn/T/workdir-host-darwin-10_14/go
# - GOPATH: /var/folders/9w/4l2_g3kx01x199n37fbmv3s80000gn/T/workdir-host-darwin-10_14/gopath
# - env: GO_BUILDER_NAME=darwin-amd64-10_14 GOROOT_BOOTSTRAP=/Users/gopher/goboot
```

Note that these lack the `/private` prefix. So I think something somewhere is forgetting to set `PWD` and/or `GOROOT`, and as a result the implicit `GOROOT` is ending up as whatever the OS reports instead. (To my recollection, on macOS `/var` is a symlink to `/private/var`.)

**Comment by bcmills** [comment]
I mailed out a CL to fix this, but I think it will require some deployment work to determine whether it fixes the problem.

**Comment by bcmills** [comment]
This builder issue is blocking the fix for #48319, which is a real bug that I would like to be able to fix for Go 1.18.

**Comment by bcmills** [comment]
Delightful! [CL 353549](https://golang.org/cl/353549) does seem to have done the trick. We could submit [CL 223755](https://golang.org/cl/223755) to prevent future regressions, but if I understand correctly it might seriously degrade the performance of certain builders (notably plan9). It was very useful for debugging but I'm indifferent as to whether we actually merge it.
